src/sentry/grouping/fingerprinting/__init__.py

```diff
@@ -108,6 +108,36 @@ class _ReleaseInfo(TypedDict):
     release: str | None
 
 
+Fingerprint = list[str]
+
+
+class FingerprintRuleAttributes(TypedDict):
+    title: NotRequired[str]
+
+
+# Specifically, a list of [<name of event attribute to match>, <value to match>]
+FingerprintMatchConfig = list[str]
+FingerprintMatchJSON = list[str]
```

Comment:
I can see where you're coming from on the indirection front, but at the same time, I feel like it's nice purely as a form of free documentation, no? Admittedly it emphasizes function over form, but IMHO the former is more important in terms of being able to read the code.

(As for the JSON thing - I hear you. I chose that name because it's what gets output from the `to_json` method.)

Reply:
please remove the aliases -- they add very little here and make it harder to read the code and understand the types